SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -552,6 +553,22 @@ void USpatialNetDriver::OnGSMQuerySuccess()
 	// If the deployment is now accepting players and we are waiting to spawn. Spawn.
 	if (bWaitingToSpawn && ClientCanSendPlayerSpawnRequests())
 	{
+		// Are we running with the same schema hash as the server?
+#if !(UE_BUILD_SHIPPING)
+		if (GetDefault<USpatialGDKSettings>()->bEnableSchemaValidationOnJoin)
+		{
+			uint32 ServerHash = GlobalStateManager->GetSchemaHash();
+			if (ClassInfoManager->SchemaDatabase->SchemaDescriptorHash != ServerHash)
+			{
+				UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Your clients Spatial schema does match the servers, this may cause problems. Client hash: '%u' Server hash: '%u'"), ClassInfoManager->SchemaDatabase->SchemaDescriptorHash, ServerHash);
+				if (FMessageDialog::Open(EAppMsgType::YesNo, EAppReturnType::No, FText::FromString(TEXT("Schema hash does not match the server, this may cause problems. Are you sure you want to continue?"))) == EAppReturnType::No)
```

Comment:
Just realised that this may cause issues for Simualted Players as they can't handle these message dialogs...
Might be worth noting down.

Reply:
Or adding a timeout!

Reply:
What will happen? Should we just log an error instead? It feels a bit intrusive to have a plugin show a dialog